src/main/java/com/rakovets/course/java/core/practice/io/FileAnalyzeUtil.java

```diff
@@ -0,0 +1,139 @@
+            }
+        }
+        return repetitionList;
+    }
+
+    //taks07
+    public static Map<String, Integer> getCountRepitingWords(Path filePath)  {
+        Map<String, Integer> repetitionOfWords = new HashMap<>();
+        String[] words = getListFromFile(filePath).toString().trim()
+                .replaceAll("[\\[\\],.!?]", "").split(" ");
+        for (int i = 0; i < words.length; i++) {
+            repetitionOfWords.put(words[i], repetitionOfWords.getOrDefault(words[i], 0) + 1);
+        }
+        return repetitionOfWords.entrySet().stream()
+                .sorted(Map.Entry.comparingByValue())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2, LinkedHashMap::new));
+    }
+
+    //task08
+    public static void getSortedNumbersInFile(Path filePath) {
+        String[] arrayNumbers = getListFromFile(filePath).toString()
+                .replaceAll("[^0-9\\s]", "").split(" ");
+        List<Integer> sortedNumbers = Arrays.stream(arrayNumbers)
+                .map(Integer::parseInt)
+                .sorted()
+                .collect(Collectors.toList());
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(filePath + "_"))) {
+            bufferedWriter.write(sortedNumbers.toString());
+            bufferedWriter.flush();
+        } catch (IOException ex) {
+            System.out.println(ex.getMessage());
+        }
+    }
+
+    //task09
+    public static double getStudentsProgress(Path filePath) {
+        String text;
+        double marksCount = 0;
+        double sumMarks = 0;
+        double averageMark = 0;
+        try (BufferedReader reader = new BufferedReader(new FileReader(String.valueOf(filePath)))) {
+            while ((text = reader.readLine()) != null) {
+                String[] array = text.trim().replaceAll("[a-zA-Z]", "").split("[, ]");
+                for (String value : array) {
+                    try {
+                        sumMarks += Double.parseDouble(value.trim());
+                        marksCount++;
+                    } catch (NumberFormatException ignored) {
+                    }
+                }
+            }
+        } catch (IOException ex) {
+            System.out.println(ex.getMessage());
+        }
+        averageMark = BigDecimal.valueOf(sumMarks / marksCount).setScale(2, RoundingMode.HALF_UP).doubleValue();
+        return averageMark;
+    }
+
+    //taks10
+    public static void getChangeModifier(String filePath, String xModifier, String yModifier) {
+        String newFilePath = filePath.replace(filePath, filePath + "_");
+        String oldJavaCode;
+        String newJavaCode;
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath));
+             BufferedWriter writer = new BufferedWriter(new FileWriter(newFilePath))) {
+            while ((oldJavaCode = reader.readLine()) != null) {
+                newJavaCode = oldJavaCode.replace(xModifier, yModifier);
```

Comment:
This isn't correct because it will rename method `privateMyNameMethod` to `publicMyNameMethod`